//*****基于FDbus的车载仪表互联通讯协议

//通讯地址："svc://cluster_fdbus"

syntax = "proto3"; //协议
package CLUSTER;  //包名

//消息ID
//根据消息ID解析对应的Message
//值为1-99的ID通过broadcast方法调用（接收方须要先行调用subscribe方法订阅）
//值为100-199的ID通过invoke或reply方法调用
//值为200-255的ID通过send方法调用
enum MsgID
{
    TYPE_UNKNOWN = 0;                     //未知消息
    
    TYPE_VERSION = 101;                   //对应【版本号消息】Version
}

//【版本号消息】
//通过reply方法调用，通过invoke查询时可以不构造该message
message Version
{
    string text = 1; //完整版本号字符串
    uint32 major_num = 2; //主版本号
    uint32 minor_num = 3; //次版本号
    uint32 revision_num = 4; //修订版本号
    bool beta = 5; //测试版本或发布版本
    string commit_id = 6; //git提交的版本号
}
